### ISSUE TITLE
Update guides for mapping 3D buildings

### ISSUE THREAD

**Issue by geografa** [mapbox/mapping#245]
tl;dr Update mapping guide to cover best practices for editing buildings.

----
Mapping buildings are covered here:

- https://www.mapbox.com/mapping/osm-data-model/#relations
- https://www.mapbox.com/mapping/mapping-common-features/#buildings

Links to other resources are here:

- http://wiki.openstreetmap.org/wiki/Simple_3D_buildings#Building_relation
- https://www.mapbox.com/blog/mapping-3d-buildings/

Some of these seem to conflict. Examples:

- does one create an outline and tag the it with `building=yes` and the total levels? Or do I only tag the relation if using relations? 
- when does one tag the relation `type=multipolygon` over `type=building`?
- for multi-purpose buildings, are POIs encouraged over adding footprints to something like first-floor storefronts?
- when creating outlines do you add height to the outline (default) building or create a `part` for this? The [Willis Tower example](https://www.mapbox.com/blog/mapping-3d-buildings/) appears to add height to both the outline and the part.

### Relations

The `type=building` relation don't appear to have an effect on the renderer. However, F4 does consider building relations in that it treats the outline way of a building relation as a mask on all the other members of the relation. How should we address relations in the guide?

@chtnha @ajashton @1ec5

**Comment by ajashton** [comment]
> does one create an outline and tag the it with building=yes and the total levels? Or do I only tag the relation if using relations?

I would recommend doing this on the building outline (either the outer way or the `type=multipolygon` relation if there is one) for maximum compatibility.

> when does one tag the relation type=multipolygon over type=building?

These are two different things and sometimes you might want both. The `type=multipolygon` would be a multipolygon relation for the building outline in cases where there is a courtyard/hole in the building - but it would not include any of the `building:part=yes` pieces. The `type=building` collects all of the different parts of a 3D building together - both the outline and the `building:part` pieces.

> when creating outlines do you add height to the outline (default) building or create a part for this? The Willis Tower example appears to add height to both the outline and the part.

In the Willis Tower example, each of the `building:part` pieces has a different height. The height assigned to the main `building=yes` outline represents the maximum height of the building, but the full outline should not be extruded to that height.

**Comment by chtnha** [comment]
@geografa Months ago I was writing a gist on [how to add 3d tags](https://gist.github.com/chtnha/8fa4c44c8cf9a12262548a98053460da) to OSM buildings. I will improve the same and add it to [our mapping guides](https://www.mapbox.com/mapping/). I welcome If someone wants to improve it and suggest ideas to make necessary changes. 

 @ajashton @1ec5

**Comment by maning** [comment]
@chtnha Status here? If no next actions, let's revisit later.
